src/test/server.c

```diff
@@ -125,6 +137,17 @@ void *pdispatch(void *data)
         printf("In server callback for msg #%ld\n", msg->hinfo.cookie);
     }
 
+    if (!authorize(msg->hinfo.target, msg->hinfo.instance))
```

Comment:
But why is a field necessary for this?  The server already saw the client's X.509 certificate so the server can keep a per-connection ID field.  The client doesn't need to send an instance ID with every request since the server knows which client it is talking to.